pkg/idutil/imagewalker/imagewalker.go

```diff
@@ -64,14 +64,26 @@ func (w *ImageWalker) Walk(ctx context.Context, req string) (int, error) {
 		return -1, err
 	}
 
-	matchCount := len(images)
 	// to handle the `rmi -f` case where returned images are different but
 	// have the same short prefix.
 	uniqueImages := make(map[digest.Digest]bool)
 	for _, image := range images {
 		uniqueImages[image.Target.Digest] = true
 	}
 
+	// Allow to nerdctl rmi <short digest ids of another images> to remove images.
+	if len(uniqueImages) > 1 {
+		imageIDPrefix := fmt.Sprintf("sha256:%s", regexp.QuoteMeta(req))
```

Comment:
I am not sure `regexp.QuoteMeta` serves a purpose here?

Reply:
Fixed